examples/sha256/table16.rs

```diff
@@ -177,74 +185,134 @@ impl<F: FieldExt> Table16Config<F> {
     }
 }
 
-impl<F: FieldExt> Config for Table16Config<F> {
+impl<F: FieldExt, L: Layouter<Field = F>> Config for Table16Config<'_, F, L> {
+    type Root = Self;
     type Field = F;
     type Configured = Table16Configured;
     type Loaded = ();
+    type Layouter = L;
+
+    fn get_root(&mut self) -> &mut Self::Root {
+        self
+    }
+
+    fn configured(&self) -> &Self::Configured {
+        &self.configured
+    }
 
-    fn load(layouter: &mut impl Layouter<Self>) -> Result<(), Error> {
-        let table = layouter.configured().lookup_table.clone();
-        table.load(layouter)
+    fn loaded(&self) -> &Self::Loaded {
+        &()
+    }
+
+    fn load(&mut self) -> Result<(), Error> {
+        let configured = self.configured().lookup_table.clone();
+        let mut spread_table = SpreadTableConfig::<'_, F, L> {
+            configured,
+            layouter: self.layouter(),
+            marker: PhantomData,
+        };
+        spread_table.load()
+    }
+
+    fn layouter(&mut self) -> &mut Self::Layouter {
+        self.layouter
+    }
+
+    fn push_namespace<NR, N>(&mut self, _name_fn: N)
+    where
+        NR: Into<String>,
+        N: FnOnce() -> NR,
+    {
+        // TODO
+    }
+
+    /// Exits out of the existing namespace.
+    ///
+    /// Not intended for downstream consumption; use [`Layouter::namespace`] instead.
+    fn pop_namespace(&mut self, _gadget_name: Option<String>) {
+        // TODO
     }
 }
 
-impl<F: FieldExt> Sha256Instructions for Table16Config<F> {
+impl<F: FieldExt, L: Layouter<Field = F>> Sha256Instructions for Table16Config<'_, F, L> {
     type State = State;
     type BlockWord = BlockWord;
 
     fn zero() -> Self::BlockWord {
         BlockWord::new(0)
     }
 
-    fn initialization_vector(layouter: &mut impl Layouter<Self>) -> Result<State, Error> {
-        let configured = layouter.configured().clone();
-        configured.compression.initialize_with_iv(layouter, IV)
+    fn initialization_vector(&mut self) -> Result<State, Error> {
+        let configured = self.configured().compression.clone();
+
+        let mut compression = CompressionConfig::<'_, F, L> {
+            configured,
+            layouter: self.layouter(),
+            marker: PhantomData,
+        };
+        compression.initialize_with_iv(IV)
```

Comment:
I had imagined that a Config would instantiate each sub-Config only once, rather than each time an instruction is used. Re-instantiating it will work because only the `Layouter` is actually (directly) mutable, and that's shared. Still, it's quite verbose and a little bit inefficient to do it this way.